cheetah/accelerator.py

```diff
@@ -80,6 +87,8 @@ def __call__(self, incoming):
             tm = self.transfer_map(incoming.energy)
             mu = torch.matmul(tm, incoming._mu)
             cov = torch.matmul(tm, torch.matmul(incoming._cov, tm.t()))
+            if self.device != "cpu":
```

Comment:
The '__ParameterBeam__' does not have a parameter '__device__', since the device for a '__ParameterBeam__' is always the __"cpu"__. It would be possible to add a attribute '__device__' to '__ParameterBeam__', which is then always set on __"cpu"__ but I think that would be quite pointless. 
It also covers all possibilities. The device for calculating the segment can be either __"cpu"__ or __"cuda"__. The device for the calculation is always __"cpu"__. Therefore if the device for calculating the segment is __"cuda"__, an error should be displayed and it is. If the device for calculating the segment is __"cpu"__, no error should be displayed and it is not.
However if a '__ParticleBeam__' is used the devices used for the calculations could also differ. I am going to implement a check for this too. Since the device for calculating a '__ParticleBeam__' can differ, i will use the device of the '__ParticleBeam__' for the comparison.